### ISSUE TITLE
Providing accessible themes

### ISSUE THREAD

**Issue by marekbrze** [saadeghi/daisyui#1174]
Most of the current Daisy UI themes don't pass contrast checking test for WCAG AA and AAA levels.

It would be good to provide few themes that pass even the strict WCAG test levels.

**Comment by saadeghi** [comment]
I think what most people are missing about the concept of a11y is that WCAG tests and their thresholds are not accurate at all. We can't just compare foreground and background colors and tell if it's accessible or not. That's just lazy.

Color accessibility depends on many factors that WCAG is not considering at all!
- A color contrast may be readable with one font family/size/weight and may be inaccessible with another font family/size/weight.
- A color contrast might look perfect on a good LED monitor but may not be readable at all on a cheap display.
- A color combination may look good for a large logo but can be hard to see on a small logo.
- You can get a perfect score of WCAG but your text may be unreadable because of line spacing or letter spacing.

It matters how and where we use the colors.

Unfortunately we can't automate these things (yet) with currently available metrics,  specially WCAG which is simply inaccurate. You can run WCAG test on the most famous web pages and get a bad result. That doesn't mean the creators of page don't care about accessibility.

For daisyUI, [I wrote a test](https://github.com/saadeghi/daisyui/blob/master/src/tests/a11y.test.js) to make sure all color combinations of all themes have a WCAG score of 3:1 at least but increasing the contrast more than that is not worth it.

My friendly advice, stop caring about WCAG number. It's not accurate. Accessibility depends on many other things that WCAG does not care about. If you really care about accessibility, you should test your specific product with real users.

Let me know if you have any questions

**Comment by sdoering01** [comment]
First of all, thank you for such a great component library that uses Tailwind. I really appreciate the effort you put into this. 

The one thing that annoyed me, was the color themes. I didn't need to look at the WCAG numbers to notice that some of the foregrounds and backgrounds don't match well. I have never had problems with such stuff in other designs, but in the standard dark theme it really feels like some colors melt/blur into each other, which makes it a hard on my eyes. This happens especially for the primary and accent colors and on the outline buttons on the landing page. Looking at the WCAG numbers just confirmed what I have experienced.
I am sadly not a designer (which is why I reach out to component frameworks and predefined themes in the first place) and can't contribute on this, but I wanted to give you this feedback, because this problem prevents me from using DaisyUI for anything that isn't a toy project. I think this is a missed opportunity for an otherwise very well designed component framework.

**Comment by saadeghi** [comment]
@sdoering01 I totally agree that `outline primary button` on a `dark` background doesn't have enough contrast but let's see why this happens?  
1. We have a `primary` color for the background color of the button. And `primary-content` color for the text color. These 2 colors have a good contrast with each other.
2. We also need any button to have a `outline` style where we have no idea what the background color would be. Our button is just transparent. And it only has text color and a border color.

That being said, the outline primary button can not have a good contrast on the dark background because the button has no idea about the background color so for an accessible color combination, we should only use the `outline primary button` on a background that is indeed readable.

Any example on daisyUI docs is just there to demonstrate how any possible combination of component variants could look like.

Also, no one is limited to only use the built-in themes. I always encourage people to make their own themes and frankly making a theme takes very little time. See the [theme generator](https://daisyui.com/theme-generator/), [colors guide](https://daisyui.com/docs/colors/) and [themes docs](https://daisyui.com/docs/themes/)

I hope my color choices for a one of a themes doesn't prevent anyone from using daisyUI because those themes are just my design decisions but themes are totally customizable.

let me know if you have any questions.
